app/src/main/java/org/mozilla/fenix/onboarding/DefaultBrowserNotificationWorker.kt

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Context.NOTIFICATION_SERVICE
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationCompat.BADGE_ICON_NONE
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import mozilla.components.support.base.ids.SharedIdsHelper
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.utils.Settings
+
+class DefaultBrowserNotificationWorker(
+    context: Context,
+    workerParameters: WorkerParameters
+) : Worker(context, workerParameters) {
+
+    override fun doWork(): Result {
+        ensureChannelExists()
+        NotificationManagerCompat.from(applicationContext)
+            .notify(NOTIFICATION_TAG, NOTIFICATION_ID, buildNotification())
+        return Result.success()
+    }
+
+    /**
+     * Build the notification with site controls to be displayed while the web app is active.
+     */
+    private fun buildNotification(): Notification {
+        val channelId = ensureChannelExists()
+        val intent = Intent(applicationContext, HomeActivity::class.java)
+        intent.putExtra(INTENT_DEFAULT_BROWSER_NOTIFICATION, true)
+
+        val pendingIntent = PendingIntent.getActivity(
+            applicationContext,
+            SharedIdsHelper.getNextIdForTag(applicationContext, NOTIFICATION_PENDING_INTENT_TAG),
+            intent,
+            0
+        )
+
+        with(applicationContext) {
+            val appName = getString(R.string.app_name)
+            return NotificationCompat.Builder(this, channelId)
+                .setSmallIcon(R.drawable.ic_status_logo)
+                .setContentTitle(
+                    applicationContext.getString(R.string.notification_default_browser_text, appName))
+                .setBadgeIconType(BADGE_ICON_NONE)
+                .setColor(ContextCompat.getColor(this, R.color.primary_text_light_theme))
+                .setPriority(NotificationCompat.PRIORITY_LOW)
+                .setShowWhen(false)
+                .setContentIntent(pendingIntent)
+                .build()
```

Comment:
Is `PRIORITY_LOW` intentional? That means, by your video as well, that the notification will never "pop-up" when it is delivered but it will hide (minimized) until the user opens up their either notification tray and find it there. That seems like the opposite desire for a notification that we want to catch the user's attention with. We might want to ask UX or marketing about that.

With `BADGE_ICON_NONE` might have similar effects of hiding the notification, but I'm not fully sure of what [the docs](https://developer.android.com/reference/android/app/Notification?hl=ja#BADGE_ICON_NONE) mean.

Reply:
This is a good question.  My assumption is that we don't want to make any noise with this notification.  However, this is just my assumption.  @topotropic, do we want this notification more visible to the user?  Thanks,

Reply:
I'll update the priority to `PRIORITY_DEFAULT` and `BADGE_ICON_SMALL` so it's more visible.  I'll upload a new recording after.  Thanks,